src/header.rs

```diff
@@ -206,7 +176,7 @@ impl TbfHeader {
         }
 
         // Return the length by generating the header and seeing how long it is.
-        self.generate().unwrap().get_ref().len()
+        self.generate().unwrap_or_else(|_| io::Cursor::new(Vec::new())).get_ref().len()
```

Comment:
This will make this line never fail, correct? Why would we want that? If the generate call cannot be unwrapped, there is no point in returning a header length of 0.

Reply:
Yeah, that is true in general I'm allergic to `raw unwrap()`!

I didn't think this through, ok If I change to `expect("Error: header is empty")`